avocado/utils/sysinfo.py

```diff
@@ -0,0 +1,386 @@
+                # all at once.
+                in_data = in_messages.read(DATA_SIZE)
+                if not in_data:
+                    break
+                yield in_data
+
+    def __eq__(self, other):
+        if hash(self) == hash(other):
+            return True
+        elif isinstance(other, Collectible):
+            return False
+        return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        return not result
+
+    def __hash__(self):
+        return hash((self.log_path, Collectible))
+
+
+class Logfile(Collectible):
+
+    """
+    Collectible system file.
+
+    :param path: Path to the log file.
+    :param log_path: Basename of the file where output is logged (optional).
+    """
+
+    def __init__(self, path, log_path=None):
+        if not log_path:
+            log_path = os.path.basename(path)
+        super(Logfile, self).__init__(log_path)
+        self.path = path
+
+    def __repr__(self):
+        r = "Logfile(%r, %r)"
+        r %= (self.path, self.log_path)
+        return r
+
+    def __eq__(self, other):
+        if isinstance(other, Logfile):
+            return (self.path, self.log_path) == (other.path, other.log_path)
+        elif isinstance(other, Collectible):
+            return False
+        return NotImplemented
+
+    def __hash__(self):
+        return hash((self.path, self.log_path, Logfile))
+
+    def collect(self):
+        """
+        Reads the log file.
+        :raise CollectibleException
+        """
+        if os.path.exists(self.path):
+            try:
+                yield from self._read_file(self.path)
+            except IOError:
+                raise CollectibleException("Not logging %s (lack of "
+                                           "permissions)" % self.path)
+        else:
+            raise CollectibleException("Not logging %s (file not found)" %
+                                       self.path)
+
+
+class Command(Collectible):
+
+    """
+    Collectible command.
+
+    :param cmd: String with the command.
+    :param timeout: Timeout for command execution.
+    :param locale: Force LANG for sysinfo collection
+    """
+
+    def __init__(self, cmd, timeout=-1, locale='C'):
+        super(Command, self).__init__(cmd)
+        self._name = self.log_path
+        self.cmd = cmd
+        self.timeout = timeout
+        self.locale = locale
+
+    def __repr__(self):
+        r = "Command(%r, %r)"
+        r %= (self.cmd, self.log_path)
+        return r
+
+    def __eq__(self, other):
+        if isinstance(other, Command):
+            return (self.cmd, self.log_path) == (other.cmd, other.log_path)
+        elif isinstance(other, Collectible):
+            return False
+        return NotImplemented
+
+    def __hash__(self):
+        return hash((self.cmd, self.log_path, Command))
+
+    def collect(self):
```

Comment:
I see you also changed some methods while moving the class to a new file. This makes it really hard to review in one commit. Is it possible to make the changes and commit them before moving the classes to a new file?

Reply:
sure I will spit the commit.